src/main/kotlin/ar/edu/unahur/obj2/semillasAlViento/Planta.kt

```diff
@@ -1,39 +1,34 @@
 package ar.edu.unahur.obj2.semillasAlViento
 
-abstract class Planta(val anioObtencionSemilla: Int, var altura: Float) {
-  fun esFuerte() = this.horasDeSolQueTolera() > 10
-
-  fun parcelaTieneComplicaciones(parcela: Parcela) =
-    parcela.plantas.any { it.horasDeSolQueTolera() < parcela.horasSolPorDia }
+abstract class Planta(val anioObtencionSemilla: Int, val altura: Float) {
 
+  fun esFuerte() = this.horasDeSolQueTolera() > 10
   abstract fun horasDeSolQueTolera(): Int
   abstract fun daSemillas(): Boolean
+  fun calcularHorasBase(altura: Float): Int{
+    return when {
+      altura < 0.5  -> 6
+      altura < 1    -> 7
+      else          -> 9
+    }
+ }
 }
 
 class Menta(anioObtencionSemilla: Int, altura: Float) : Planta(anioObtencionSemilla, altura) {
   override fun horasDeSolQueTolera() = 6
   override fun daSemillas() = this.esFuerte() || altura > 0.4
```

Comment:
Para no repetir la parte de `this.esFuerte()` en todas las subclases, dos opciones:
1. Ponerlo en la superclase y luego hacer `super.daSemillas() || altura > 0.4`.
2. Hacer un template method: un método auxiliar abstracto `condicionEspecie` en `Planta` que luego cada subclase tenga que definir.